### PR TITLE
wrapGAppsHook fixes

### PR DESCRIPTION
**Pull request by seam345**
I have a found a few packages that crash when attempting to open file pickers with the following error
```
GLib-GIO-ERROR **: 06:51:14.382: No GSettings schemas are installed on the system
```

###### Description of changes

Fixed in the same way for all 3 following what was recommended in #16285

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Other

I was unsure if this should be 3 separate PR's as they are separate applications if this is preferred, I am happy to split them up.

### CODE REVIEW

File: pkgs/applications/editors/texmaker/default.nix [hunk 1]
```diff
@@ -9,7 +9,7 @@ mkDerivation rec {
     sha256 = "0qczc2r01vhap11xmqizwbq21ggn4yjrxim8iqjxaq9w1rg2x9dz";
   };
 
-  buildInputs = [ qtbase qtscript poppler zlib qtwebengine ];
+  buildInputs = [ qtbase qtscript poppler zlib qtwebengine wrapGAppsHook ];
   nativeBuildInputs = [ pkg-config poppler qmake ];
```
Comment (wegank) [inline pkgs/applications/editors/texmaker/default.nix:13]:
```suggestion
  buildInputs = [ qtbase qtscript poppler zlib qtwebengine ];
  nativeBuildInputs = [ pkg-config poppler qmake wrapGAppsHook ];
```

Comment (seam345) [reply pkgs/applications/editors/texmaker/default.nix]: Opps thanks for spotting :)

Updated in latest push https://github.com/NixOS/nixpkgs/pull/227005/files#diff-6d83d8c41bdb1c85cf61904fbcc3d39b9f4fc7d1e46a9f7143b800c9a626a737R13

**Comment by AndersonTorres** [comment]
I am a bit afraid of double wrapping. But LGTM.

**Comment by seam345** [comment]
Yeah I was initially surprised to see `wrapQtAppsHook` in vlc but opening the file chooser still persisted the error and (I'll be honest) i never checked what the wrapQt does differently/ or otherwise but adding the `wrapGAppsHook` worked (for me) so i just bulldozed ahead :D

File: pkgs/applications/video/vlc/default.nix
Comment (Artturin) [inline pkgs/applications/video/vlc/default.nix:1]: do
```
  dontWrapGApps = true;

  preFixup = ''
    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
  '';
```

to not doublewrap

Comment (seam345) [reply pkgs/applications/video/vlc/default.nix:1]: Added in 769e70bd3e1 wasn't 100% where best to put it so put it above postFixup, happy to move it elsewhere if preferred.

File: pkgs/applications/editors/texmaker/default.nix
Comment (Artturin) [inline pkgs/applications/editors/texmaker/default.nix:1]: texmaker and spectacle both need double wrap fix

qt mkDerivation is adding wrapQt

```
nix-repl> texmaker.nativeBuildInputs
[ «derivation /nix/store/5l92g2q5ymaii3pz7s36y88madg6rs54-pkg-config-wrapper-0.29.2.drv» «derivation /nix/store/s27nw99313gbwh0dpxvl549aj56ani2p-poppler-qt5-23.02.0.drv» «derivation /nix/store/w39gpnn1swnil2jll4k2k6v9l1gsngh5-qmake-hook.drv» «derivation /nix/store/77i8vv3r7y76q7cxz3zgbd1h22s3xgpd-wrap-qt5-apps-hook.drv» ]
```

Comment (seam345) [reply pkgs/applications/editors/texmaker/default.nix:1]: ohhh oki, thanks for spotting, I'll also go back and check Hugin #217828 with that command

**Comment by seam345** [comment]
opps accidentally caught another commit in my rebase, fixing now

File: pkgs/applications/editors/texmaker/default.nix
Comment (seam345) [reply pkgs/applications/editors/texmaker/default.nix:1]: Added to Spectacle and Texmaker now :)

**Comment by seam345** [comment]
rebased to fix merge conflict on spectacle

**Pull request merged**